web/packages/teleport/src/services/integrations/integrations.ts

```diff
@@ -292,10 +293,24 @@ export const integrationService = {
       .then(resp => resp.serviceDashboardUrl);
   },
 
-  async createAwsAppAccess(integrationName): Promise<App> {
-    return api
-      .post(cfg.getAwsAppAccessUrl(integrationName), null)
-      .then(makeApp);
+  async createAwsAppAccess(
+    integrationName,
+    req: CreateAwsAppAccessRequest
+  ): Promise<App> {
+    // TODO(kimlisa): DELETE IN 19.0 - replaced by v2 endpoint.
+    if (!req.labels || !Object.keys(req.labels).length) {
+      return api
+        .post(cfg.getAwsAppAccessUrl(integrationName), req)
+        .then(makeApp);
+    }
+
+    return (
+      api
+        .post(cfg.getAwsAppAccessUrlV2(integrationName), req)
+        .then(makeApp)
+        // TODO(kimlisa): DELETE IN 19.0
+        .catch(withUnsupportedLabelFeatureErrorConversion)
+    );
```

Comment:
i think the new UI should always use the new endpoint regardless if labels exist or not. The only time an "old" endpoint should be used is if we want some sort of graceful fallback in some feature which we don't care about here.

go all in!

Reply:
i might be just confused, but i added the fallback so that the user has the option to proceed without the labels if they keep hitting the old proxy?

Reply:
> But this block of code reads like it will hit v1 if they didnt add any labels.

i think that's okay, since either endpoint in new proxy uses the same handler

> I think they should always hit v2 even without labels

hrm, but what if they hit an old proxy? get an error, and they get stuck? (the fallback is then to just remove labels and it will use the old endpoint in case it hits the old proxy again)

Reply:
i mean they should always hit v2 as the happy path, even if they dont want to add labels. they are optional, so they should always go v2 since v2 should be able to handle it.
if the web hits the 404, they can be given the option to fallback to v1, sure. but we should never be hitting v1 if we havent already got the 404

Reply:
ah i think i finally understand the problem. calling v1 without labels will have issues once we remove the old endpoint, great catch 👍